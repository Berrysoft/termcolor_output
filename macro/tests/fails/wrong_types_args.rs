use termcolor::NoColor;
use termcolor_output as tco;

struct NonDebuggable;

fn main() {
    let mut w: NoColor<Vec<u8>> = NoColor::new(vec![]);
    match tco::write!(w, "{}{}{:?}", Some(()), vec![1u8], NonDebuggable) {
        Ok(_) => {}
        Err(_) => {}
    };
}
