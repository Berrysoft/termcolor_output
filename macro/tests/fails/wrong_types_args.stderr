error[E0277]: `Option<()>` doesn't implement `std::fmt::Display`
 --> $DIR/wrong_types_args.rs:8:35
  |
8 |     match colored!(w, "{}{}{:?}", Some(()), vec![1u8], NonDebuggable) {
  |                                   ^^^^^^^^ `Option<()>` cannot be formatted with the default formatter
  |
  = help: the trait `std::fmt::Display` is not implemented for `Option<()>`
  = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
  = note: required by `std::fmt::Display::fmt`
  = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: `Vec<u8>` doesn't implement `std::fmt::Display`
 --> $DIR/wrong_types_args.rs:8:45
  |
8 |     match colored!(w, "{}{}{:?}", Some(()), vec![1u8], NonDebuggable) {
  |                                             ^^^^^^^^^ `Vec<u8>` cannot be formatted with the default formatter
  |
  = help: the trait `std::fmt::Display` is not implemented for `Vec<u8>`
  = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
  = note: required by `std::fmt::Display::fmt`
  = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: `NonDebuggable` doesn't implement `Debug`
 --> $DIR/wrong_types_args.rs:8:56
  |
8 |     match colored!(w, "{}{}{:?}", Some(()), vec![1u8], NonDebuggable) {
  |                                                        ^^^^^^^^^^^^^ `NonDebuggable` cannot be formatted using `{:?}`
  |
  = help: the trait `Debug` is not implemented for `NonDebuggable`
  = note: add `#[derive(Debug)]` or manually implement `Debug`
  = note: required by `std::fmt::Debug::fmt`
  = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
